third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -40,15 +40,16 @@ Value redundantDataMask(Type valueTy, ConversionPatternRewriter &rewriter,
     auto sizePerThread = triton::gpu::getSizePerThread(layout);
     auto threadsPerWarp = triton::gpu::getThreadsPerWarp(layout);
     auto warpsPerCTA = triton::gpu::getWarpsPerCTA(layout);
-    auto order = triton::gpu::getOrder(layout);
+    auto threadOrder = triton::gpu::getThreadOrder(layout);
+    auto warpOrder = triton::gpu::getWarpOrder(layout);
```

Comment:
What if you call `getWarpOrder(dot operand layout)` here?

Reply:
IMO it seems just fine to still have `getWarpOrder` defined for dot operand layouts.

But it causes confusion for you, I will add a condition for dot operand layouts here.

Reply:
In that case it will break hard. That is good, because incidentally `delinearize` would not work as expected. Before it would create incorrect results, now at least it breaks hard.

Note that there is no combination of `warpsPerCTA`  and `warpOrder` that can represent the order of DotOperands, so I think it's better to break hard,